juniper-vpn.py

```diff
@@ -234,7 +237,8 @@ def action_key(self):
                 sys.exit(1)
             self.key = hotp(self.args.oath)
         elif self.key is None:
-            self.key = getpass.getpass('Two-factor key:')
+            matrix = self.get_matrix_index(self.br.response().read())
+            self.key = getpass.getpass('Two-factor key (' + matrix + '):')
```

Comment:
I am pretty sure it is. Probably when the regex does not match, we know there is no Matrixcard auth. But there should be better ways to determine. Unfortunately, I do not have any possibillity to try this out, because the only available server here is either Private Certificate based, or Matrixcard. No time based token! :(